### PR TITLE
Feature/jfas/environment warning history clean

### LINKED ISSUE #158: Add environment warning on receipt-popup

### PR DESCRIPTION
**Pull request by jonasanker**
Closes #158 

Cleanup of dirty commit history in #181

### CODE REVIEW

**Comment by Jens-Fastrup** [comment]
Having issues on this branch (as well as origin/develop) loading MobilePay reciepts, stats, settings. Set up the mobilepay test app with login(nr, pin, cpr)  but still does not bother. What did I miss (also did a full run build)?

**Comment by TTA777** [comment]
> Having issues on this branch (as well as origin/develop) loading MobilePay reciepts, stats, settings. Set up the mobilepay test app with login(nr, pin, cpr) but still does not bother. What did I miss (also did a full run build)?

Since this is a bit light on details, I'll give guess, but I am not sure. If it is everything that you are unable to load data from, try logging out of the app. If you are unable to access the log out button on the settings page, due to this page not loading, you can clear the apps data cache using the android settings. 

I'm guessing what has happened is you were signed in to another backend previously, so your app still thinks you are signed in, since it still has a token cached, which is what we now want to clear.

**Comment by Jens-Fastrup** [comment]
Sounds very plausible! Will try those steps

**Comment by marfavi** [comment]
Will review

File: lib/cubits/environment/environment_state.dart [hunk 1]
```diff
@@ -1,9 +1,13 @@
 part of 'environment_cubit.dart';
 
-enum Environment { unknown, test, production }
+abstract class EnvironmentState {}
 
-extension EnvironmentIs on Environment {
-  bool get isUnknown => this == Environment.unknown;
-  bool get isTest => this == Environment.test;
-  bool get isProduction => this == Environment.production;
+class EnvironmentInitial extends EnvironmentState {}
```
Comment (fremartini) [inline lib/cubits/environment/environment_state.dart:5]: change to EnvironmentLoading to stay consistent with other states?

File: lib/widgets/components/receipt/receipt_overlay.dart [hunk 2]
```diff
@@ -13,30 +13,31 @@ class ReceiptOverlay {
...
-                      receipt.transactionType == TransactionType.purchase,
-                  isInOverlay: true,
-                ),
-                Text(
-                  Strings.receiptTapAnywhereToDismiss,
-                  style: AppTextStyle.explainerBright,
-                )
-              ],
-            ),
+        return Center(
```
Comment (marfavi) [inline lib/widgets/components/receipt/receipt_overlay.dart:25]: Seems like the padding that should wrap this widget is removed.

**Review by marfavi — CHANGES_REQUESTED** [review]

File: lib/widgets/components/receipt/receipt_overlay.dart (hunk 2 above)
Comment (marfavi) [reply lib/widgets/components/receipt/receipt_overlay.dart:25]: image

**Comment by fremartini** [comment]
@Jens-Fastrup can you have a look at finishing this one?

**Comment by Jens-Fastrup** [comment]
@fremartini , spoke with @marfavi today who will add the padding as I have issues with building the app on my desktop. He will likely resolve it soon

**Comment by marfavi** [comment]
Added padding & made it more responsive as well (less padding on smaller screens)
image

**Review by marfavi — APPROVED** [review]

**Pull request merged**